recipes-extended/xen/files/libxl-display-manager-support.patch

```diff
@@ -46,26 +46,40 @@ PATCHES
  static int
  libxl__xc_device_get_rdm(libxl__gc *gc,
                           uint32_t flags,
-@@ -929,6 +937,7 @@ static int libxl__build_device_model_arg
+@@ -934,6 +942,7 @@ static int libxl__build_device_model_arg
      const int num_disks = guest_config->num_disks;
      const int num_nics = guest_config->num_nics;
      const libxl_vnc_info *vnc = libxl__dm_vnc(guest_config);
 +    const libxl_display_info *display = libxl__dm_display(guest_config);
      const libxl_sdl_info *sdl = dm_sdl(guest_config);
      const char *keymap = dm_keymap(guest_config);
      char *machinearg;
-@@ -1049,7 +1058,10 @@ static int libxl__build_device_model_arg
-     /*
-      * OpenXT: the default display backend is Surfman
-      */
--    flexarray_append_pair(dm_args, "-display", "surfman");
-+    if (display && display->kind)
-+        flexarray_append_pair(dm_args, "-display", display->kind);
-+    else
-+        flexarray_append_pair(dm_args, "-display", "surfman");
+@@ -1048,11 +1057,6 @@ static int libxl__build_device_model_arg
+         flexarray_append(dm_args, vncarg);
+     } /* OpenXT: no else here, we don't support "-vnc none" */
  
+-    /*
+-     * OpenXT: the default display backend is Surfman
```

Comment:
Seems like you're removing code from another patch here, wouldn't it make more sense to edit that other patch instead?

Reply:
libxl-display-manager-support.patch was already touching code on top of libxl-openxt-qemu-args.patch.  My patches were developed on top of both, so they naturally modify the topmost one.

I could redo libxl-openxt-qemu-args.patch to move the hunks and then redo libxl-display-manager-support.patch on top.  Or rework the patch queue further to first add "display->kind" and then add the desired final hunk. That would be weird since the patch would add a field without a user.  If we always make display->kind default to "surfman", then we can avoid the conditional.

Do you want one of these changes, and do you have a preference which one?

Reply:
Maybe we can rework that in a later PR, this alone is not enough of a concern to delay merging the whole thing.